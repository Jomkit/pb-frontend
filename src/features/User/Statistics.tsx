import { useEffect, useState } from 'react';
import Api from '../../api';
import { Duration } from 'luxon';

const Statistics = ({userId}: {userId: number}) => {
  const [loading, setLoading] = useState(true);
  const [allProjects, setAllProjects] = useState([] as any[]);
  
  useEffect(() => {
    const getReport = async () => {
      if(!userId) return;

      const allProjectsCall = await Api.getUserProjects(userId);
      
      setAllProjects(allProjectsCall);
      console.log("All projects", allProjectsCall);
    }

    getReport();
    setLoading(false);
  }, [userId]);
  
  return (
    <div>
      <div>My Statistics</div>
        {loading ? <p>Loading...</p> :
          <table className='table-auto w-full simple-border'>
            <thead className='bg-green-200 simple-border'>
              <tr>
                <th>Project</th>
                <th>Status</th>
                <th>Total Time Spent</th>
              </tr>
            </thead>
            <tbody>
              {allProjects.map((project: any) => (
              <tr key={project.id}>
                <td>{project.name}</td>
                <td>{project.archived ? "Complete" : "Active"}</td>
                <td>{Duration.fromISO(project.duration).toHuman()}</td>
              </tr>
                
              ))}
            </tbody>
          </table>
        }
    </div>
  )
}

export default Statistics