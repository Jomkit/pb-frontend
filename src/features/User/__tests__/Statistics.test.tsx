import { render } from "@testing-library/react"
import Statistics from "../Statistics"

beforeEach(() => {
    vi.mock("../../../api");
})

afterEach(() => {
    vi.restoreAllMocks();
})

it("should render without crashing", () => {
    render(<Statistics userId={1} />)
})

it("should match snapshot", async () => {
    const { asFragment, findByText } = render(<Statistics userId={1} />);

    const testProject1 = await findByText("testProject 1");
    expect(testProject1).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
})