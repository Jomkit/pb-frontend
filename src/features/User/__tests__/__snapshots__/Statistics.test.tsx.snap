// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      My Statistics
    </div>
    <table
      class="table-auto w-full simple-border"
    >
      <thead
        class="bg-green-200 simple-border"
      >
        <tr>
          <th>
            Project
          </th>
          <th>
            Status
          </th>
          <th>
            Total Time Spent
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            testProject 1
          </td>
          <td>
            Active
          </td>
          <td>
            5 minutes
          </td>
        </tr>
        <tr>
          <td>
            testProject 2
          </td>
          <td>
            Active
          </td>
          <td>
            5 minutes
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
